ethics.py

```diff
@@ -70,18 +69,25 @@ def findEthicalValueOfCell(self, cell):
                     neighborCellValue = neighbor.decisionModelFactor * ((extent * certainty * proximity) * ((intensity + duration) + (discount * (futureIntensity + futureDuration))))
                 else:
                     neighborCellValue = neighbor.decisionModelFactor * ((certainty * proximity) * ((extent * (intensity + duration)) + (discount * (futureExtent * (futureIntensity + futureDuration)))))
-            if selfishnessFactor > 0:
+
+            if self.decisionModelTribalFactor >= 0:
+                if neighbor.findTribe() == self.findTribe():
+                    neighborCellValue *= self.decisionModelTribalFactor
+                else:
+                    neighborCellValue *= 1 - self.decisionModelTribalFactor
+            if self.selfishnessFactor > 0:
```

Comment:
By the way, this should be `>= 0` for Altruist models.

Reply:
Very true! Patch it in.